### PR TITLE
facelift for traces, added errors

### PR DESCRIPTION
**Pull request by debris**
changes are described here: https://github.com/ethcore/parity/issues/1722#issuecomment-244059382

### CODE REVIEW

File: rpc/src/v1/types/trace.rs [hunk 1]
```diff
@@ -530,26 +589,26 @@ mod tests {
...
+			block_number: 13.into(),
+			block_hash: 14.into(),
 		};
 		let serialized = serde_json::to_string(&t).unwrap();
-		assert_eq!(serialized, r#"{"action":{"call":{"from":"0x0000000000000000000000000000000000000004","to":"0x0000000000000000000000000000000000000005","value":"0x6","gas":"0x7","input":"0x1234","callType":"call"}},"result":{"call":{"gasUsed":"0x8","output":"0x5678"}},"traceAddress":["0xa"],"subtraces":"0x1","transactionPosition":"0xb","transactionHash":"0x000000000000000000000000000000000000000000000000000000000000000c","blockNumber":"0xd","blockHash":"0x000000000000000000000000000000000000000000000000000000000000000e"}"#);
+		assert_eq!(serialized, r#"{"type":"call","action":{"from":"0x0000000000000000000000000000000000000004","to":"0x0000000000000000000000000000000000000005","value":"0x6","gas":"0x7","input":"0x1234","callType":"call"},"result":{"gasUsed":"0x8","output":"0x5678"},"traceAddress":["0xa"],"subtraces":"0x1","transactionPosition":"0xb","transactionHash":"0x000000000000000000000000000000000000000000000000000000000000000c","blockNumber":"0xd","blockHash":"0x000000000000000000000000000000000000000000000000000000000000000e"}"#);
```
Comment (tomusdrw) [inline rpc/src/v1/types/trace.rs]: Additional tests would be nice (especialy for errors)

Comment (debris) [inline rpc/src/v1/types/trace.rs]: sure 👍

**Comment by arkpar** [comment]
@debris could you please also updated the documentation?

**Comment by debris** [comment]
done https://gist.github.com/debris/051c131e877affefeab4553509640a43

**Pull request merged**